src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+}
+```
+
+**Example**: annotate directly on the schema
+```pdl
+@data_classification = "HIGH"
+record UserPersonallyIdentifiableInformation {
+  firstName: string,
+  lastName: string,
+  userId: long,
+  socialSecurityNumber: string,
+}
+```
+Note that these annotations are on the schema level.
+
+The support that Rest.Li provided for processing annotation in Rest.Li schema has been quite limited. These annotations in above examples are later stored as a `DataSchema`'s class attribute called "property". Just as the example shows, both field and the DataSchema can have this "property". We let the user chooses where they want to have the schema annotated and how they want to interpret. There were no preferrabled way regarding how annotation should be written, read and interpreted.
+
+## Inherit and override schema annotations
+But Rest.Li users found it useful to process annotations during schema processing. One use case is to introduce "inheritance" and "overrides" to annotations so those annotations can be dynamically processed in the way user defines, when for example the schemas were reused. This gives annotation extensibility and adds to schema reusability.
+
+Here are examples:
+Example case 1: Users might want the annotation of a field be inherited. The fields can reuse the annotations from upper level.
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, we might find both `visitedUrls` and `visitedUserProfiles` have `persistencePolicyInDays` annotated as 30.
+//<inheritance example>
+
+Example case 2: Users might want the annotation of a field be overriden. Override is where the annotation on a field or a schema might get updated when other annotations assign it another value.
+//<override example>
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, a user might find the field `visitedUserProfiles`s `persistencePolicyInDays` annotation is 30 but `visitedUrls`'s `persistencePolicyInDays` annotation has value 10, which overrides the original value inherited(which was 30).
+
+
+Example case 3:  Override might also happen when some annotation needs to be updated by another value assigned from another annotation location
+```pdl
+record UserVisitRecord {
+  //...
+  @persistencePolicyInDays = 365
+  recycledChatHistories: array[chat]
+
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  //...
+}
+
+record EnterpriseUserRecord {
+  userName: UserName
+  //...
+  @persistencePolicyInDays = {"/recycledChatHistories" : 3650}
+  visitRecord: UserVisitRecord
+  //...
+}
+```
+In this example, schema `EnterpriseUserRecord` reused `UserVisitRecord` and its annotation in the field `visitRecord`, but it overrides the annotation value of `recycledChatHistories`.
+
+
```

Comment:
nits:
- `examples shows` -> `examples show`
- `overrides gives` -> `overrides give`
- `to the schema annotations` -> `to schema annotations`
- `how those annotations can be read` -> `how those annotations are read`
- `imambiguous` -> `unambiguous`
- `user's rules` -> `the user's rules`
- `call such value "resolved" value` -> `call this value the "resolved" value`